src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java

```diff
@@ -103,7 +107,10 @@ final class GrpcServiceStubs implements IGrpcServiceStubs {
     headers.put(RPC_ENCODING_HEADER_KEY, "proto");
     Channel interceptedChannel =
         ClientInterceptors.intercept(
-            channel, deadlineInterceptor, MetadataUtils.newAttachHeadersInterceptor(headers));
+            channel,
+            deadlineInterceptor,
+            MetadataUtils.newAttachHeadersInterceptor(headers),
+            newOpenTelemetryInterceptor());
```

Comment:
is this going to be created per request basis? or a single interceptor will be reused?

Reply:
This interceptor is used for interceptedChannel instance creation which is created per GrcpServiceStubs instance.